prov/util/src/ipc_handle_monitor.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * (C) Copyright (c) 2022 Amazon.com, Inc. or its affiliates.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "ofi_mr.h"
+#include "ofi_hmem.h"
+
+static int ipc_handle_monitor_subscribe(struct ofi_mem_monitor *monitor,
+					 const void *addr, size_t len,
+					 union ofi_mr_hmem_info *hmem_info)
+{
+	/* no-op */
+	return FI_SUCCESS;
+}
+
+static void ipc_handle_monitor_unsubscribe(struct ofi_mem_monitor *monitor,
+					   const void *addr, size_t len,
+					   union ofi_mr_hmem_info *hmem_info)
+{
+	/* no-op */
+}
+
+static bool ipc_handle_monitor_valid(struct ofi_mem_monitor *monitor,
+			const struct ofi_mr_info *info,
+			struct ofi_mr_entry *entry)
+{
+	return (memcmp((void **)&info->ipc_handle,
+		(void **)&entry->info.ipc_handle,
+		IPC_HANDLE_SIZE) == 0);
+}
+
+static int ipc_handle_monitor_start(struct ofi_mem_monitor *monitor)
+{
+	/* no-op */
+	return FI_SUCCESS;
+}
+
+void ipc_handle_monitor_stop(struct ofi_mem_monitor *monitor)
+{
+	/* no-op */
+}
+
+static struct ofi_mem_monitor ipc_handle_monitor = {
+	.init = ofi_monitor_init,
+	.cleanup = ofi_monitor_cleanup,
+	.start = ipc_handle_monitor_start,
+	.stop = ipc_handle_monitor_stop,
+	.subscribe = ipc_handle_monitor_subscribe,
+	.unsubscribe = ipc_handle_monitor_unsubscribe,
+	.valid = ipc_handle_monitor_valid,
+};
```

Comment:
ipc_handle_monitor works fine for now, but I believe we will want hmem iface specific monitors.  (E.g. cuda_ipc_monitor, ze_ipc_monitor, etc.)  The only function out of these that does anything is the valid() call, which is technically an iface check specific.  The other functions should be declared as generic no-op calls, which are not specific to an ipc monitor.  That is, rename ipc_handle_monitor_start -> ofi_mm_start_noop, etc.  Other implementations that are no-ops can simply reference these no-op calls (which should move into a better location).  For example, you can do ```#define rocr_mm_unsubscribe ofi_mm_unsubscribe_noop``` where the definition isn't needed.

See below.

Reply:
OK,  I can start from renaming `ipc_handle_monitor.c` to `cuda_ipc_monitor.c`, implement the general no-ops function, and then create `ze/rocr_ipc_monitor.c` and make the valid() as `ENOSYS` for now.